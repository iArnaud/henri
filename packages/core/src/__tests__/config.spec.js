const henri = require('../henri');
const Config = require('../config');
const BaseModule = require('../base/module');

describe('config', () => {
  beforeEach(() => {
    require('../index');
  });

  test('should be accessible from henri', () => {
    expect(henri.config).toBeDefined();
    expect(henri.config.name).toBe('config');
  });

  test('loads data', () => {
    const data = henri.config.get('abc');
    expect(data).toEqual('c');
  });

  test('should suppress warnings', () => {
    expect(process.env.SUPPRESS_NO_CONFIG_WARNING).toBeTruthy();
  });

  test('inherits from BaseModule', () => {
    const newConfig = new Config();
    expect(newConfig).toBeInstanceOf(Config);
    expect(newConfig).toBeInstanceOf(BaseModule);
    expect(henri.config.reload).toBeDefined();
    expect(henri.config.init).toBeDefined();
    expect(henri.config.setup).toBeDefined();
    expect(henri.config.start).toBeDefined();
    expect(henri.config.stop).toBeDefined();
  });

  test('it reloads', () => {
    expect(henri.config.reloadable).toBeTruthy();
    henri.config.reload();
  });

  test('has a safe mode', () => {
    expect(() => henri.config.get('unknown key')).toThrow();
    expect(() => henri.config.get('unknown key', true)).not.toThrow();
    expect(henri.config.get('unknown key', true)).toBeFalsy();
  });
});
