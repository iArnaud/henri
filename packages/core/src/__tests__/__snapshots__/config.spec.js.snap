// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config in test should match snapshot 1`] = `
Config {
  "config": null,
  "consoleOnly": false,
  "get": [Function],
  "has": [Function],
  "henri": null,
  "info": [Function],
  "init": [Function],
  "key": null,
  "name": "config",
  "reload": [Function],
  "reloadable": true,
  "runlevel": 0,
  "setup": [Function],
  "start": [Function],
}
`;
